diff --git a/services/src/bpamio.c b/services/src/bpamio.c
index 8a5464a..879937c 100644
--- a/services/src/bpamio.c
+++ b/services/src/bpamio.c
@@ -1032,7 +1032,9 @@ int deq_dataset_member(const char* ds, const char* wmem, const DBG_Opts* opts)
 
 record_format_t record_format(const FM_BPAMHandle* bh, const DBG_Opts* opts)
 {
-  if ((bh->dcb->dcbexlst.dcbrecfm & dcbrecf) && (bh->dcb->dcbexlst.dcbrecfm & dcbrecbr)) {
+  if ((bh->dcb->dcbexlst.dcbrecfm & dcbrecu) == dcbrecu) {
+    return RECORD_FORMAT_U;
+  } else if ((bh->dcb->dcbexlst.dcbrecfm & dcbrecf) && (bh->dcb->dcbexlst.dcbrecfm & dcbrecbr)) {
     return RECORD_FORMAT_FB;
   } else if (bh->dcb->dcbexlst.dcbrecfm & dcbrecf) {
     return RECORD_FORMAT_F;
@@ -1040,8 +1042,6 @@ record_format_t record_format(const FM_BPAMHandle* bh, const DBG_Opts* opts)
     return RECORD_FORMAT_VB;
   } else if (bh->dcb->dcbexlst.dcbrecfm & dcbrecv) {
     return RECORD_FORMAT_V;
-  } else if (bh->dcb->dcbexlst.dcbrecfm & dcbrecu) {
-    return RECORD_FORMAT_U;
   }
   return RECORD_FORMAT_UNKNOWN;
 }
